### DISCUSSION TITLE
Is it possible to conditionally obfuscate a specific field in a collection?

### DISCUSSION THREAD

**Discussion by GiorgioBetta** [pocketbase/pocketbase#810] [Q&A]
Trips Collection is public but:
If Trip.user.shareWithPublic is false => Trip information are still public but Trip.price is obfuscated and instead of it's value is returned by the api as a chosen value like "xxxx.xx" ? 
Is it possible to achieve something like this? I'm specifically interested in the substitution of the field and not in hiding it from the response.

**Comment by ganigeorgiev** [comment]
You can change the API response only with the PocketBase event hooks - https://pocketbase.io/docs/event-hooks/ (in your case that will be the [Record API hooks](https://pocketbase.io/docs/event-hooks/#record-hooks))

**Comment by MyraManes** [comment]
I was able to accomplish something similar on my users table by creating a view using the following query:

`SELECT id, first_name, last_name, (CASE WHEN is_phone_publicly_available THEN users.phone_number ELSE '000-000-0000' END) AS phone_number, (CASE WHEN is_email_publicly_available THEN users.email ELSE 'xxxxxxxxxx@xxxxx.xxx' END) AS email FROM users;`
